okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java

```diff
@@ -227,7 +228,15 @@ public Level getLevel() {
         Charset charset = UTF8;
         MediaType contentType = responseBody.contentType();
         if (contentType != null) {
-          charset = contentType.charset(UTF8);
+          try {
+            charset = contentType.charset(UTF8);
+          } catch (UnsupportedCharsetException e) {
+            logger.log("");
+            logger.log("Couldn't decode the response body; charset is likely malformed.");
```

Comment:
If they got this far they'll see the header logged above.

On Sat, Jan 16, 2016, 6:53 PM Jesse Wilson <notifications@github.com> wrote:

> In
> okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java
> <https://github.com/square/okhttp/pull/2255#discussion_r49938948>:
>
> > @@ -227,7 +228,15 @@ public Level getLevel() {
> >          Charset charset = UTF8;
> >          MediaType contentType = responseBody.contentType();
> >          if (contentType != null) {
> > -          charset = contentType.charset(UTF8);
> > +          try {
> > +            charset = contentType.charset(UTF8);
> > +          } catch (UnsupportedCharsetException e) {
> > +            logger.log("");
> > +            logger.log("Couldn't decode the response body; charset is likely malformed.");
>
> maybe include the MediaType so the unlucky user can see UFT-8 or whatever
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/okhttp/pull/2255/files#r49938948>.
>